### PR TITLE
Add build support for ChromeOS platform

### PR DESCRIPTION
**Pull request by RavirajSitaram**
Modify cmake files to prevent downloading libraries from
third part servers.

Signed-off-by: P Raviraj Sitaram <raviraj.p.sitaram@intel.com>

### Details:
 - *item1*
 - *...*

### Tickets:
 - EISW-37847

### CODE REVIEW

File: cmake/developer_package/linux_name.cmake [hunk 1]
```diff
@@ -6,17 +6,22 @@ include(target_flags)
...
 if (LINUX)
     function(get_linux_name res_var)
-        if (NOT EXISTS "/etc/lsb-release")
-            execute_process(COMMAND find -L /etc/ -maxdepth 1 -type f -name *-release -exec cat {} \;
-                    OUTPUT_VARIABLE release_data RESULT_VARIABLE result)
-            string(REPLACE "Red Hat" "CentOS" release_data "${release_data}")
-            set(name_regex "NAME=\"([^ \"\n]*).*\"\n")
-            set(version_regex "VERSION=\"([0-9]+(\\.[0-9]+)?)[^\n]*\"")
-        else ()
+        if (${TARGET_OS_NAME} STREQUAL "CHROMIUMOS")
```
Comment (ilya-lavrenov) [inline cmake/developer_package/linux_name.cmake:9]: `${TARGET_OS_NAME}` => `TARGET_OS_NAME`

File: cmake/developer_package/packaging.cmake [hunk 2]
```diff
@@ -20,9 +20,17 @@ function(ie_cpack_set_library_dir)
...
-        set(IE_CPACK_LIBRARY_PATH runtime/lib/${ARCH_FOLDER} PARENT_SCOPE)
-        set(IE_CPACK_RUNTIME_PATH runtime/lib/${ARCH_FOLDER} PARENT_SCOPE)
-        set(IE_CPACK_ARCHIVE_PATH runtime/lib/${ARCH_FOLDER} PARENT_SCOPE)
+        if (NOT LINUX_OS_NAME STREQUAL "CHROMIUMOS")
+            set(IE_CPACK_LIBRARY_PATH runtime/lib/${ARCH_FOLDER} PARENT_SCOPE)
+            set(IE_CPACK_RUNTIME_PATH runtime/lib/${ARCH_FOLDER} PARENT_SCOPE)
+            set(IE_CPACK_ARCHIVE_PATH runtime/lib/${ARCH_FOLDER} PARENT_SCOPE)
+            set(IE_CPACK_INCLUDES_PATH runtime/include PARENT_SCOPE)
+        else()
+            set(IE_CPACK_LIBRARY_PATH ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR} PARENT_SCOPE)
```
Comment (ilya-lavrenov) [inline cmake/developer_package/packaging.cmake:29]: - do we need explicit `CMAKE_INSTALL_PREFIX` here?
- we need to include https://cmake.org/cmake/help/v3.0/module/GNUInstallDirs.html to have `CMAKE_INSTALL_LIBDIR` available 
- can we also use `CMAKE_INSTALL_INCLUDEDIR`?

File: src/plugins/intel_gna/cmake/libGNAConfig.cmake [hunk 3]
```diff
@@ -36,7 +36,12 @@ if(libGNA_FIND_REQUIRED_KERNEL)
 endif()
 
 if(libGNA_FIND_REQUIRED_API)
-    find_path(libGNA_INCLUDE_DIRS gna2-api.h PATHS "${GNA_EXT_DIR}/include")
+    if(NOT LINUX_OS_NAME STREQUAL "CHROMIUMOS")
+        find_path(libGNA_INCLUDE_DIRS gna2-api.h PATHS "${GNA_EXT_DIR}/include")
+    else()
+        find_path(libGNA_INCLUDE_DIRS gna2-api.h PATHS ${GNA_INCLUDE_LOC})
```
Comment (ilya-lavrenov) [inline src/plugins/intel_gna/cmake/libGNAConfig.cmake:42]: where `GNA_INCLUDE_LOC` is defined?

File: cmake/dependencies.cmake [hunk 4]
```diff
@@ -26,22 +26,29 @@ fetch_models_and_validation_set()
...
+        find_host_program(
+            SYSTEM_PROTOC
+            NAMES protoc
+            PATHS "${SYSTEM_PROTOC_ROOT}/bin"
+            NO_DEFAULT_PATH)
+    else()
+        find_host_program(
+            SYSTEM_PROTOC
+            NAMES protoc
+            PATHS "${SYSTEM_PROTOC_ROOT}/bin")
```
Comment (ilya-lavrenov) [inline cmake/dependencies.cmake:49]: who defines `SYSTEM_PROTOC_ROOT` in this case?

File: cmake/dependencies.cmake [hunk 5]
```diff
@@ -26,22 +26,29 @@ fetch_models_and_validation_set()
 get_linux_name(LINUX_OS_NAME)
 
 if(CMAKE_CROSSCOMPILING AND CMAKE_HOST_SYSTEM_NAME MATCHES Linux AND CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "amd64.*|x86_64.*|AMD64.*")
```
Comment (ilya-lavrenov) [inline cmake/dependencies.cmake:28]: am I right that compiling for `CHROMIUMOS` is a cross-compilation? should we create a cmake toolchain for this case? Like here https://github.com/openvinotoolkit/openvino/tree/master/cmake/toolchains

File: cmake/dependencies.cmake [hunk 6]
```diff
@@ -281,6 +288,11 @@ if(ENABLE_OPENCV)
             elseif((LINUX_OS_NAME STREQUAL "Ubuntu 20.04" OR LINUX_OS_NAME STREQUAL "LinuxMint 20.1") AND X86_64)
                 set(OPENCV_SUFFIX "ubuntu20")
                 set(OPENCV_HASH "2fe7bbc40e1186eb8d099822038cae2821abf617ac7a16fadf98f377c723e268")
+            elseif(LINUX_OS_NAME STREQUAL "CHROMIUMOS" AND X86_64)
+                find_package(OpenCV)
+                if (NOT OpenCV_FOUND)
```
Comment (ilya-lavrenov) [inline cmake/dependencies.cmake:293]: using this command you will find system OpenCV, which is Linux x86 one. Why not to use OpenCV corresponding to [CMKAE_HOST_SYSTEM](https://cmake.org/cmake/help/v3.0/variable/CMAKE_HOST_SYSTEM.html)?

File: src/plugins/intel_gna/cmake/libGNAConfig.cmake (hunk 3 above)
Comment (RavirajSitaram) [reply src/plugins/intel_gna/cmake/libGNAConfig.cmake:42]: GNA_INCLUDE_LOC will be provided in ebuild file. https://chromium.googlesource.com/chromiumos/docs/+/HEAD/portage/ebuild_faq.md#ebuild-faq

File: cmake/dependencies.cmake (hunk 4 above)
Comment (RavirajSitaram) [reply cmake/dependencies.cmake:49]: chromeos sdk provides an older version of protoc library.  However if the newer version is required, goal is to pass the location of package (SYSTEM_PROTOC_ROOT) as cmake variable using ebuilds (https://chromium.googlesource.com/chromiumos/docs/+/HEAD/portage/ebuild_faq.md#ebuild-faq)

If protoc is only used by front end libraries (Paddle / Tf / Onnx), can we download only if front end libraries are enabled like here.. 
https://github.com/openvinotoolkit/openvino/blob/9dee25fa79e771c7a4a5759c6021bb2af687d292/thirdparty/CMakeLists.txt#L85

File: cmake/developer_package/linux_name.cmake (hunk 1 above)
Comment (RavirajSitaram) [reply cmake/developer_package/linux_name.cmake]: Updated the code with change

File: cmake/developer_package/packaging.cmake (hunk 2 above)
Comment (RavirajSitaram) [reply cmake/developer_package/packaging.cmake]: Updated the code with the changes

**Comment by akladiev** [comment]
This PR will be closed in 2 weeks in case of no activity.

**Comment by akladiev** [comment]
This PR was closed because it has been stalled for 2 week with no activity.

**Pull request closed without merging**